internal/validators/base.go

```diff
@@ -0,0 +1,53 @@
+package validators
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"path/filepath"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	apimachyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Validator interface {
+	Validate(fs.FS) error
+}
+
+// TODO(asmacdo) I moved this *and* made it public, is this a problem?
+func GetObjects(bundleFS fs.FS) ([]client.Object, error) {
```

Comment:
Do we use this anywhere outside the plain provisioner (I think not since it has knowledge and makes assumptions about the FS that's passed in)?

If it is specific to the plain provisioner, I think it should be internal and within the plain package tree.

Reply:
bundle deployment uses it as well https://github.com/operator-framework/rukpak/blob/0f45b79342c53aab4e1ee4dee1958f7c95e3e422/internal/provisioner/plain/controllers/bundledeployment_controller.go#L436